launcher/src/test/java/org/apache/brooklyn/launcher/CleanOrphanedLocationsTest.java

```diff
@@ -138,6 +175,14 @@ public void testKeepsLocationsReferencedByEnricher() throws Exception {
     }
     
     @Test
+    public void testKeepsLocationsReferencedByEnricherInFlag() throws Exception {
+        Location loc = mgmt().getLocationManager().createLocation(LocationSpec.create(SshMachineLocation.class));
+        origApp.enrichers().add(EnricherSpec.create(MyEnricher.class)
```

Comment:
This doesn't store the location in the flag - it's just stored as a config key. `MyEnricher.obj` needs to change to `@SetFromFlag("myobj") Object obj;`